source/dsl/GitDSL.ts

```diff
@@ -64,8 +76,7 @@ export interface GitDSL {
    * @param {string} filename the path to the json file
    * @param {string[]} diffTypes An array containing: "add", "del" or "normal" - then you can only get these changes
    */
-
-  diffForFile(filename: string, diffTypes?: string[]): string | null,
+  diffForFile(filename: string, diffTypes?: string[]): Promise<TextDiff | null>,
```

Comment:
I honestly am confused about the meaning of `normal` in this context -- I _think_ it refers to "contextual" lines (i.e. non-changed lines that sandwich an addition/deletion). I have a hard time finding why it'd be useful to have these separately without the benefit of combining them with either `added` or `removed`.

Maybe we could just provide the raw `changes` structure from `parseDiff`'s output as an escape hatch for folks who need finer control.